doc/UserGuide.md

```diff
@@ -468,32 +468,39 @@ value ultimately reported as failing might not satisfy the condition given to
 case that does. To avoid confusion, the corresponding shrink for the type can
 use `suchThat` method too.
 
-#### Generating Containers
+#### Generating Buildables and Containers
 
-There is a special generator, `Gen.containerOf`, that generates containers such
-as lists and arrays. They take another generator as argument, that is
-responsible for generating the individual items. You can use it in the
-following way:
+There are special generators
+* `Gen.buildableOf`, that generates any buildable collection
+* `Gen.containerOf`, a convenience method for single parameter buildable like lists and arrays
+
+They take another generator as argument, that is responsible for generating the individual items.
+You can use it in the following way:
 
 ```scala
-val genIntList        = Gen.containerOf[List,Int](Gen.oneOf(1, 3, 5))
+implicit genInt: Gen[Int] = Gen.oneOf(1, 3, 5)
+val genIntList            = Gen.containerOf[List, Int]
```

Comment:
`Gen.containerOf` takes the nested `Gen` value as an explicit parameter:
https://github.com/typelevel/scalacheck/blob/c8bd5ae8b1b014003f00576a0912d0fbe0fceee3/core/shared/src/main/scala/org/scalacheck/Gen.scala#L996-L999

Moreover, I feel the `Gen` type is not supposed to be passed around implicitly. The `Arbitrary` one is for that.